src/switcher.S

```diff
@@ -0,0 +1,134 @@
+#include "comps_offsets.h"
+
+.text
+.balign 4
+
+.global switcher_entry
+.global switch_compartment
+.global switch_compartment_end
+
+/**
+ * Entry point from user code to switcher function
+ *
+ * @param c0 DDC of switcher, containing compartment information
+ */
+.type switcher_entry, "function"
+switcher_entry:
+    mov      c29, c0
+    mov      x0, #0
+    cvtp     clr, lr
+    b        switch_compartment
+    ret      clr
```

Comment:
Do we ever come back here? The `b` looks like a tail call, so is this instruction necessary?

Reply:
Ah, good point. We can just drop through. I copy pasted this code from the previous switcher iteration (and honestly shouldn't have included it in this PR), but for completeness, I updated it in https://github.com/capablevms/cheri-morello-compartmentalisation/pull/1/commits/08f834ff048f627173e5959fa7fe7d54a2107c66.